### ISSUE TITLE
‘\t’  display messy code

### ISSUE THREAD

**Issue by hugo-hong** [szieke/ScriptCommunicator_serial-terminal#32]

**Comment by szieke** [comment]
Hi,

what is '\t'? Is this a tab (hex 0x9)? And what do you mean with 'messy code?

**Comment by hugo-hong** [comment]
Yes, for example, when used printf(“\tLink Type: 0x01 -> ACL\n”); output log, it will show that:



 

Best Regards

  _____  

洪永光｜Hugo Hong

**Comment by szieke** [comment]
ScriptCommunicator prints an empty line or this?
ÿLink Type: 0x01 -> ACL

ScriptCommunicator should print ÿ instead of '\t' because it prints an ÿ for every non-printable character (except for \r and \n) in the ASCII console. 
But I think it is a good idea to print a tab for a '\t' in the ASCII console. I will add this in the next version of ScriptCommunicator.

**Comment by hugo-hong** [comment]
Thanks for your replay.

 

And I have the following suggestions:

1.      Add utf8 charset console.

2.      Add Tab page to open one more serial ports instead of open another ScriptCommunicator instance. 

 

Best Regards

  _____  

洪永光｜Hugo Hong

**Comment by szieke** [comment]
1. I think I will add this in a future release. 
You can find a worker script that displays all received data in a utf8 console under exampleScripts\WorkerScripts\MainWindowConsole\Utf8Console

2.If you need a second interface you can use a worker script. You can find an example under exampleScripts\WorkerScripts\MultiSerialPort.

Best Regards

**Comment by hugo-hong** [comment]
Would you consider adding split screen feature? 
Just link SecritCRT split one more serial port console in a instance.

**Comment by szieke** [comment]
This would not fit into the design of ScriptCommunicator with its multi-interface (serial port, sockets, CAN ...), multi-console (ascii, hex ...), and log file functionality. But as I wrote, you can extend ScriptCommunicator with worker scripts for your use case.

**Comment by hugo-hong** [comment]
OK, use the exampleScripts\WorkerScripts\MainWindowConsole\Utf8Console, can show utf8 console, very good!

But there is another issue: it cannot append timestamp in the utf8 console.

**Comment by szieke** [comment]
The utf8 script is able to generate time stamps after a time period. For that you have to:
- enable the option 'create time stamp after ms' (settings window/console option
- start the script (the script reads the options only once at the start)

**Comment by hugo-hong** [comment]
yes, i have enable create time stamp at receive \n char, the ASCII console can append timestamp, but the utf8 cannot.

**Comment by szieke** [comment]
I modified the script. Now it adds a timestamp at a specified character (attachment). But I discovered a bug in the ScriptCommunicatir API. At the moment scripts cannot read the character at which a timestamp shall be created. Therefore you have to modify the character in the script (line 63).

**Comment by hugo-hong** [comment]
ok, replace the new utf8Console.js, the timestamp can show now, very good!!!
But the utf8 console timestamp have no color code and ascii with.  And i suggest add utf8 console to setting option better.

**Comment by szieke** [comment]
I will add a utf8 console in the next release. Then won't ne the script anymore.
